test/esm-named-exports.test.js

```diff
@@ -59,6 +59,7 @@ const ignoredCjsKeysMap = {
     'VIRTUAL',
   ],
   '@sequelize/core/decorators-legacy': ['__esModule'],
+  // '@sequelize/mssql': ['__esModule'],
```

Comment:
I've commented this out as I was getting the following error when it was running.

```shell
Error: global leak(s) detected: '__extends', '__assign', '__rest', '__decorate', '__param', '__esDecorate', '__runInitializers', '__propKey', '__setFunctionName', '__metadata', '__awaiter', '__generator', '__exportStar', '__createBinding', '__values', '__read', '__spread', '__spreadArrays', '__spreadArray', '__await', '__asyncGenerator', '__asyncDelegator', '__asyncValues', '__makeTemplateObject', '__importStar', '__importDefault', '__classPrivateFieldGet', '__classPrivateFieldSet', '__classPrivateFieldIn', '__addDisposableResource', '__disposeResources'
```

Reply:
We should find out the cause of this, as we shouldn't be creating new globals

Reply:
The issue seems to come from tslib https://github.com/microsoft/tslib/issues/32

Reply:
Just adding the following to .mocharc.json should do the trick
```
  "global": ["__extends", "__assign", "__rest", "__decorate", "__param", "__esDecorate", "__runInitializers", "__propKey", "__setFunctionName", "__metadata", "__awaiter", "__generator", "__exportStar", "__createBinding", "__values", "__read", "__spread", "__spreadArrays", "__spreadArray", "__await", "__asyncGenerator", "__asyncDelegator", "__asyncValues", "__makeTemplateObject", "__importStar", "__importDefault", "__classPrivateFieldGet", "__classPrivateFieldSet", "__classPrivateFieldIn", "__addDisposableResource", "__disposeResources"],
```
Do check if this is the same list as #17207 or that you'll need to add some more

Reply:
Thanks @WikiRik this works a treat.

Do you want me to add a comment and reference the tslib issue?
Mocha does support JSONC so I can add it in if you want.

Reply:
Can you actually reference this discussion instead? So we remember that it is added for mssql and snowflake, and not all dialects.
Link is https://github.com/sequelize/sequelize/pull/17206#discussion_r1538541573

Reply:
Added now.